### PR TITLE
3410 sync basic user information

### LINKED ISSUE #3410: Sync basic user information
## Is your feature request related to a problem? Please describe 👀
Currently, transactions that get migrated from a desktop or mobile site have placeholder names for who created the transaction.
Screenshot 2024-03-25 at 13 56 59

## Describe the solution you'd like 🎁
Sync basic user information. 

### Describe alternatives you've considered 💭
🤷🏻 

### Moneyworks Jobcode 🧰
OMS:MIGRT

### PR DESCRIPTION
**Pull request by roxy-dao**
Fixes #3410

# 👩🏻‍💻 What does this PR do? 
Syncs basic user information to omSupply so unknown user isn't shown on invoices, stocktakes and requisitions that were transferred from OG.

# 🧪 How has/should this change been tested? 
- [ ] Run tests OR
- [ ] Use an existing OG data file
- [ ] Sync
- [ ] Side panel user information should show even if it isn't the user who logged in

### CODE REVIEW

File: server/repository/src/db_diesel/user_row.rs [hunk 1]
```diff
@@ -36,13 +36,7 @@ pub enum Language {
     Tetum,
 }
 
-impl Default for Language {
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/user_row.rs:39]: Sidetracked: Refactoring

File: server/repository/src/db_diesel/user_row.rs [hunk 2]
```diff
@@ -54,24 +48,7 @@ pub struct UserAccountRow {
     pub last_name: Option<String>,
     pub phone_number: Option<String>,
     pub job_title: Option<String>,
-    pub last_successful_sync: NaiveDateTime,
-}
-
-impl Default for UserAccountRow {
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/user_row.rs:60]: Sidetrack: Refactoring

File: server/repository/src/db_diesel/user_row.rs [hunk 3]
```diff
@@ -54,24 +48,7 @@ pub struct UserAccountRow {
...
-            email: None,
-            language: Default::default(),
-            first_name: None,
-            last_name: None,
-            phone_number: None,
-            job_title: None,
-            last_successful_sync: Defaults::naive_date_time(),
-        }
-    }
+    pub last_successful_sync: Option<NaiveDateTime>,
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/user_row.rs:51]: Made this optional so users who have never logged in before can just have a default None value for last_successful_sync

File: server/repository/src/db_diesel/user_row.rs [hunk 4]
```diff
@@ -157,20 +134,17 @@ impl<'a> UserAccountRowRepository<'a> {
     }
 }
 
-// TODO
-// Users don't sync and will only be available after first log in, thus in schema reference is not enforced
-// API consumers would like users to be returned for records that are linked to them, as if reference was enforced
-// Using uknown user until we start syncing users
-pub fn unknown_user() -> User {
```
Comment (roxy-dao) [inline server/repository/src/db_diesel/user_row.rs:164]: Removed since no longer necessary

**Review by clemens-msupply — APPROVED** [review]
Nice! and works fine. Thanks!

**Pull request merged**